### PR TITLE
feat: expand comment-level options and update documentation for PR comments

### LINKED ISSUE #102: Add new setting driving the comments on module measure values.
## Background
Users might be interested to see the modules' coverage statistics in PR comments.

## Feature
Create a new setting driving the comments that are output from the action.

## Proposed Solution
Setting name:
'comment-level'

Setting options:

* `none` - no comment generated at all
* `minimal` - just one overall statistics comment
* `changed` - comment regarding statistics of modules that
* `failing` - comment regarding statistics of modules that are failing
* `change and failing` - comment regarding statistics of modules that  have changed files or are failing
* `detailed` - comment on each module detected/defined

## Proposed Extra
Deprecate the complicated or confusing settings of
* `skip-unchanged` (or maybe not, depending on #103)
* `sensitivity`
* `comment-mode`

### PR DESCRIPTION
**Pull request by miroslavpojer**
## Overview

Implements task 30 for #102 by expanding `comment-level` from the current binary output model to the full six-level option set.

This PR adds support for:

- `none`
- `minimal`
- `full`
- `changed`
- `failed`
- `failed-or-changed`

Behavior summary:

- `none` suppresses PR comment posting entirely
- `minimal` keeps the global summary table only
- `full` keeps the current full comment output
- `changed` filters group/report/file rows to changed coverage only
- `failed` filters group/report/file rows to failing rows only
- `failed-or-changed` shows the union of changed and failing rows

The change also updates input validation, comment rendering, unit tests, and user-facing documentation.

## RLS Notes

- Added four new `comment-level` values: `none`, `changed`, `failed`, and `failed-or-changed`
- `comment-level: none` now suppresses GitHub PR comment creation/update
- Filtered comment levels keep the global summary table and only filter lower-level rows
- Added coverage for empty-result edge cases and the full `skip-unchanged × comment-level` matrix
- Updated action input docs and README examples to reflect the expanded option set

## Validation

- `python3 -m pytest tests/unit/test_action_inputs.py tests/unit/test_skip_unchanged.py tests/unit/generator/test_pr_comment_generator.py -q`
- `python3 -m pytest --cov=. tests/ --cov-fail-under=80`
- `python3 -m black --check $(git ls-files '*.py')`
- `python3 -m mypy .`

Known repo-level QA status:

- Full `pylint` is still blocked by pre-existing findings outside the scope of this PR

## Related

Closes #102 

Release Notes:

- Expand `comment-level` to six supported values for more precise PR comment control
- Allow users to suppress comments entirely with `comment-level: none`
- Support focused PR comments for changed rows, failing rows, or both
- Document the new comment-level behavior in the action metadata and README

### CODE REVIEW

File: jacoco_report/generator/pr_comment_generator.py [hunk 1]
```diff
@@ -375,22 +505,74 @@ def generate_changed_files_table_without_baseline(
...
+                if not coverage.changed_files_passed.get(file_path, True)
+            }
+            if not failed_files:
+                continue
+
+            filtered_coverage = EvaluatedReportCoverage(coverage.name, coverage.group_name)
+            filtered_coverage.overall_passed = coverage.overall_passed
+            filtered_coverage.avg_changed_files_passed = coverage.avg_changed_files_passed
+            filtered_coverage.overall_coverage_reached = coverage.overall_coverage_reached
+            filtered_coverage.avg_changed_files_coverage_reached = coverage.avg_changed_files_coverage_reached
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply jacoco_report/generator/pr_comment_generator.py]: Fixed.

File: jacoco_report/generator/pr_comment_generator.py [hunk 2]
```diff
@@ -51,33 +50,108 @@ def generate(self) -> None:
                 existing_comment = comment
                 break
 
+        if ActionInputs.get_comment_level() == CommentLevelEnum.NONE:
+            if existing_comment and ActionInputs.get_update_comment():
+                self.gh.delete_comment(existing_comment["id"])
+            return
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply jacoco_report/generator/pr_comment_generator.py]: Fixed.

File: tests/unit/generator/test_pr_comment_generator.py [hunk 3]
```diff
@@ -464,4 +526,266 @@ def test_calculate_baseline_group_diffs_no_data_returns_zero(pr_comment_generato
...
+    pr_comment_generator.gh.delete_comment.assert_not_called()
+
+
+def test_none_comment_content_is_title_only(pr_comment_generator, mocker):
+    _configure_generator_for_comment_tests(pr_comment_generator, mocker, comment_level="none")
+    _set_mixed_comment_level_fixture(pr_comment_generator)
+
+    title, body = pr_comment_generator._get_comment_content()
+
+    assert title == "**JaCoCo**"
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply tests/unit/generator/test_pr_comment_generator.py]: Fixed.

File: README.md [hunk 4]
```diff
@@ -232,6 +236,10 @@ Each entry is a YAML mapping with:
 
 #### Customizing the Comment Level
 
+##### No Comment
+
+- When the `comment-level` is set to `none`, no PR comment is posted.
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply README.md:241]: Fixed.

File: jacoco_report/generator/pr_comment_generator.py [hunk 5]
```diff
@@ -294,6 +422,7 @@ def _generate_reports_table_with_baseline(self, p: str, f: str, **kwargs) -> str
         return s
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply jacoco_report/generator/pr_comment_generator.py]: Fixed.

File: jacoco_report/generator/pr_comment_generator.py [hunk 6]
```diff
@@ -259,16 +374,29 @@ def _generate_reports_table_without_baseline(self, p: str, f: str, **kwargs) ->
 
         return s
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply jacoco_report/generator/pr_comment_generator.py]: Fixed.

File: jacoco_report/generator/pr_comment_generator.py [hunk 7]
```diff
@@ -51,33 +53,107 @@ def generate(self) -> None:
                 existing_comment = comment
                 break
 
-        if existing_comment and ActionInputs.get_update_comment():
+        if comment_level == CommentLevelEnum.NONE:
+            if existing_comment and update_comment:
+                self.gh.delete_comment(existing_comment["id"])
+            return
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply jacoco_report/generator/pr_comment_generator.py:63]: Fixed.
